### PR TITLE
Stop bad robots with qq email addresses

### PR DESCRIPTION
**Pull request by bevanloon**
paired with @thomasleese 

We are receiving large amounts of spam. This should hopefully alleviate
it by not sending support tickets when the email matches a particular
regex.

### CODE REVIEW

File: app/models/contact_ticket.rb [hunk 1]
```diff
@@ -68,6 +70,10 @@ def validate_mail_name_connection
     end
   end
 
+  def email_qq
+    /[0-9]+@qq.com/.match?(email)
```
Comment (barrucadu) [inline app/models/contact_ticket.rb]: Did you intend an exact match here?

```
irb(main):004:0> /[0-9]+@qq.com/.match? 'abc1@qq.com'
=> true
irb(main):005:0> /[0-9]+@qq.com/.match? 'abc1@qq.com.foo'
=> true
```

Comment (bevanloon) [reply app/models/contact_ticket.rb]: good point - think it's better in 37e23a7

that regex "book" you posted in slack earlier was somewhat prescient.

File: app/models/contact_ticket.rb [hunk 2]
```diff
@@ -68,6 +70,10 @@ def validate_mail_name_connection
     end
   end
 
+  def email_qq
+    /[0-9]+@qq.com$/.match?(email)
```
Comment (barrucadu) [inline app/models/contact_ticket.rb]: This still permits `abc1@qq.com`, if that's not what you want.

Comment (bevanloon) [reply app/models/contact_ticket.rb]: yep - not sure to be honest - I've not seen the data. I'm tempted to go with this as a blanket for now and adjust if we need to.

**Pull request merged**